### PR TITLE
gorm: Setup initial project structure, schema, and db connection

### PR DESCRIPTION
**Pull request by nvanbenschoten**

### CODE REVIEW

File: go/gorm/main.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+		// Create an Order.
+		{
+			tx := db.Begin()
+
+			var product Product
+			tx.First(&product, "name = ?", "P2")
+
+			var customer Customer
+			tx.First(&customer, "name = ?", "Fred")
```
Comment (RaduBerinde) [inline go/gorm/main.go]: [nit] might as well use `p2`, `fred` if you already have them as variables

Comment (nvanbenschoten) [inline go/gorm/main.go]: Yeah, this all gets deleted in the next commit anyway.

**Comment by nvanbenschoten** [comment]
Thanks for looking this over @RaduBerinde! 

I just added a second commit that introduces a really rough (not intended to merge) implementation of how I envision us performing automated testing across various ORMs. This is touched upon briefly in the README. Basically each application would run a thin http server over its respective ORM. The main reasons I decided upon http communication is because this needs to be language agnostic, and because an http server sitting directly above an ORM seems like a common use case. Our test harness can then prod from both the database and http sides to verify that the ORM is doing the right thing. I'd love to get people's thoughts on this approach, and if they think same level of testing could be achieved in a simpler way.

**Comment by RaduBerinde** [comment]
I like the idea of using http. There aren't many options given that we want to support ORMs for various languages.

**Pull request merged**